### ISSUE TITLE
DatabaseException: Expected  RegularPlannerQuery(...) Instead, got:  RegularPlannerQuery(...)

### ISSUE THREAD

**Issue by frant-hartm** [neo4j/neo4j#10091]
Executing following query (regardless the data in the database)

```
MATCH (n:`User`) 
MATCH (n)-[r0:`RATED`]->(m0)
WHERE r0.`stars` = { `ratings_stars_0` } 
WITH DISTINCT n 
SKIP 0 LIMIT 2 
RETURN n,[ [ (n)-[r_r1:`RATED`]->(m1:`Movie`) | [ r_r1, m1 ] ], [ (n)-[r_f1:`FRIENDS`]-(u1:`User`) | [ r_f1, u1 ] ] ], ID(n)
```

fails with

```
Caused by: org.neo4j.driver.v1.exceptions.DatabaseException: Expected 
RegularPlannerQuery(QueryGraph(Set(PatternRelationship(IdName(r0),(IdName(n),IdName(m0)),OUTGOING,List(RelTypeName(RATED)),SimplePatternLength)),Set(IdName(n), IdName(m0)),Set(),Selections(Set(Predicate(Set(IdName(r0)),In(Property(Variable(r0),PropertyKeyName(stars)),ListLiteral(List(Parameter(ratings_stars_0,Any))))), Predicate(Set(IdName(n)),HasLabels(Variable(n),List(LabelName(User)))))),Vector(),Set(),Set(),List()),DistinctQueryProjection(Map(n -> Variable(n)),QueryShuffle(List(),Some(SignedDecimalIntegerLiteral(0)),Some(SignedDecimalIntegerLiteral(2)))),Some(RegularPlannerQuery(QueryGraph(Set(),Set(),Set(IdName(n)),Selections(Set()),Vector(),Set(),Set(),List()),RegularQueryProjection(Map(n -> Variable(n), [ [ (n)-[r_r1:`RATED`]->(m1:`Movie`) | [ r_r1, m1 ] ], [ (n)-[r_f1:`FRIENDS`]-(u1:`User`) | [ r_f1, u1 ] ] ] -> ListLiteral(List(PatternComprehension(None,RelationshipsPattern(RelationshipChain(NodePattern(Some(Variable(n)),List(),None),RelationshipPattern(Some(Variable(r_r1)),List(RelTypeName(RATED)),None,None,OUTGOING,false),NodePattern(Some(Variable(m1)),List(LabelName(Movie)),None))),None,ListLiteral(List(Variable(r_r1), Variable(m1))),Set(Variable(n))), PatternComprehension(None,RelationshipsPattern(RelationshipChain(NodePattern(Some(Variable(n)),List(),None),RelationshipPattern(Some(Variable(r_f1)),List(RelTypeName(FRIENDS)),None,None,BOTH,false),NodePattern(Some(Variable(u1)),List(LabelName(User)),None))),None,ListLiteral(List(Variable(r_f1), Variable(u1))),Set(Variable(n))))), ID(n) -> FunctionInvocation(Namespace(List()),FunctionName(ID),false,Vector(Variable(n)))),QueryShuffle(List(),None,None)),None))) 
Instead, got: 
RegularPlannerQuery(QueryGraph(Set(PatternRelationship(IdName(r0),(IdName(n),IdName(m0)),OUTGOING,List(RelTypeName(RATED)),SimplePatternLength)),Set(IdName(n), IdName(m0)),Set(),Selections(Set(Predicate(Set(IdName(n)),HasLabels(Variable(n),List(LabelName(User)))), Predicate(Set(IdName(r0)),In(Property(Variable(r0),PropertyKeyName(stars)),ListLiteral(List(Parameter(ratings_stars_0,Any))))))),Vector(),Set(),Set(),List()),DistinctQueryProjection(Map(n -> Variable(n)),QueryShuffle(List(),None,None)),Some(RegularPlannerQuery(QueryGraph(Set(),Set(),Set(IdName(n)),Selections(Set()),Vector(),Set(),Set(),List()),RegularQueryProjection(Map(n -> Variable(n), [ [ (n)-[r_r1:`RATED`]->(m1:`Movie`) | [ r_r1, m1 ] ], [ (n)-[r_f1:`FRIENDS`]-(u1:`User`) | [ r_f1, u1 ] ] ] -> ListLiteral(List(PatternComprehension(None,RelationshipsPattern(RelationshipChain(NodePattern(Some(Variable(n)),List(),None),RelationshipPattern(Some(Variable(r_r1)),List(RelTypeName(RATED)),None,None,OUTGOING,false),NodePattern(Some(Variable(m1)),List(LabelName(Movie)),None))),None,ListLiteral(List(Variable(r_r1), Variable(m1))),Set(Variable(n))), PatternComprehension(None,RelationshipsPattern(RelationshipChain(NodePattern(Some(Variable(n)),List(),None),RelationshipPattern(Some(Variable(r_f1)),List(RelTypeName(FRIENDS)),None,None,BOTH,false),NodePattern(Some(Variable(u1)),List(LabelName(User)),None))),None,ListLiteral(List(Variable(r_f1), Variable(u1))),Set(Variable(n))))), ID(n) -> FunctionInvocation(Namespace(List()),FunctionName(ID),false,Vector(Variable(n)))),QueryShuffle(List(),None,None)),None)))
    at org.neo4j.driver.internal.net.SocketResponseHandler.handleFailureMessage(SocketResponseHandler.java:83)
    at org.neo4j.driver.internal.messaging.PackStreamMessageFormatV1$Reader.unpackFailureMessage(PackStreamMessageFormatV1.java:470)
    at org.neo4j.driver.internal.messaging.PackStreamMessageFormatV1$Reader.read(PackStreamMessageFormatV1.java:431)
    at org.neo4j.driver.internal.net.SocketClient.receiveOne(SocketClient.java:191)
    at org.neo4j.driver.internal.net.SocketClient.receiveAll(SocketClient.java:185)
    at org.neo4j.driver.internal.net.SocketConnection.receiveAll(SocketConnection.java:203)
    at org.neo4j.driver.internal.net.SocketConnection.sync(SocketConnection.java:157)
    at org.neo4j.driver.internal.net.ConcurrencyGuardingConnection.sync(ConcurrencyGuardingConnection.java:137)
    at org.neo4j.driver.internal.net.pooling.PooledSocketConnection.sync(PooledSocketConnection.java:157)
    at org.neo4j.driver.internal.ExplicitTransaction.close(ExplicitTransaction.java:115)
    at org.neo4j.ogm.drivers.bolt.transaction.BoltTransaction.commit(BoltTransaction.java:74)
    ... 30 more
```

- Neo4j version: 3.3.0-beta01 (works fine with 3.2.5)
- Operating system: Ubuntu 16.04
- API/Driver: java driver 1.4.3, 1.5.0-alpha01, browser

**Comment by frant-hartm** [comment]
This seems to be fixed on Neo4j 3.3.0-beta02.

Feel free to close.

**Comment by sherfert** [comment]
Yes, it seems fixed.
